src/crewai/knowledge/source/base_file_knowledge_source.py

```diff
@@ -1,17 +1,20 @@
 from pathlib import Path
-from typing import Union, List
+from typing import Union, List, Dict, Any
 
 from pydantic import Field
 
 from crewai.knowledge.source.base_knowledge_source import BaseKnowledgeSource
-from typing import Dict, Any
+from crewai.utilities.logger import Logger
 from crewai.knowledge.storage.knowledge_storage import KnowledgeStorage
 
 
 class BaseFileKnowledgeSource(BaseKnowledgeSource):
```

Comment:
As I'm reviewing this, I think this should be an abstract class`class BaseFileKnowledgeSource(BaseKnowledgeSource, ABC):